pyvista/plotting/plotting.py

```diff
@@ -1484,14 +1512,19 @@ def add_volume(self, volume, scalars=None, clim=None, resolution=None,
                                     reset_camera=reset_camera, name=next_name,
                                     ambient=ambient, categories=categories, loc=loc,
                                     culling=culling, clim=clim,
-                                    mapper=mapper, pickable=pickable)
+                                    mapper=mapper, pickable=pickable,
+                                    opacity_unit_distance=opacity_unit_distance,
+                                    shade=shade, diffuse=diffuse, specular=specular,
+                                    specular_power=specular_power)
 
                 actors.append(a)
             return actors
 
         if not isinstance(volume, pyvista.UniformGrid):
             raise TypeError('Type ({}) not supported for volume rendering at this time. Use `pyvista.UniformGrid`.')
 
+        if opacity_unit_distance is None:
+            opacity_unit_distance = volume.length / (np.mean(volume.dimensions) - 1)
```

Comment:
This is really close to what ParaView does (found by trial and error). I can't find the code for how ParaView chooses this, but for a lot of meshes (when the spacing is 1), this provides the same result as ParaView - I can't figure out exactly how to incorporate the spacing into this equation to get the result that ParaView has.